pkg/service/egress.go

```diff
@@ -275,17 +315,43 @@ func (s *EgressService) StopEgress(ctx context.Context, req *livekit.StopEgressR
 	if err := EnsureRecordPermission(ctx); err != nil {
 		return nil, twirpAuthError(err)
 	}
-	if s.rpcClient == nil {
-		return nil, ErrEgressNotConnected
+
+	if !s.useNewRPC {
+		s.useNewRPC = useNewEgressRPC(s.es)
+		if (s.useNewRPC && s.client == nil) || (!s.useNewRPC && s.clientDeprecated == nil) {
+			return nil, ErrEgressNotConnected
+		}
 	}
 
-	info, err := s.rpcClient.SendRequest(ctx, &livekit.EgressRequest{
-		EgressId: req.EgressId,
-		Request: &livekit.EgressRequest_Stop{
-			Stop: req,
-		},
-	})
+	var info *livekit.EgressInfo
+	var err error
+	returned := make(chan struct{})
+	go func() {
+		if s.useNewRPC {
+			info, err = s.client.StopEgress(ctx, req.EgressId, req)
+		} else {
+			// TODO: remove in future version
+			info, err = s.clientDeprecated.SendRequest(ctx, &livekit.EgressRequest{
+				EgressId: req.EgressId,
+				Request: &livekit.EgressRequest_Stop{
+					Stop: req,
+				},
+			})
+		}
+	}()
+
+	existing, loadErr := s.es.LoadEgress(ctx, req.EgressId)
+	if loadErr == nil {
+		if existing.Status != livekit.EgressStatus_EGRESS_STARTING &&
+			existing.Status != livekit.EgressStatus_EGRESS_ACTIVE {
+			return nil, fmt.Errorf("egress with status %s cannot be stopped", info.Status.String())
+		}
+	}
+	<-returned
```

Comment:
Do we ever send an object in this channel?

Reply:
nope